canvas_modules/common-canvas/src/common-properties/actions/action-factory.js

```diff
@@ -29,22 +31,32 @@ export default class ActionFactory {
 
 	generateAction(key, action) {
 		if (action) {
-			if (action.actionType === "button") {
+			if (action.actionType === ActionType.BUTTON) {
 				return (
 					<ButtonAction
 						key={"action." + key}
 						action={action}
 						controller={this.controller}
 					/>
 				);
-			} else if (action.actionType === "image") {
+			} else if (action.actionType === ActionType.IMAGE) {
 				return (
 					<ImageAction
 						key={"action." + key}
 						action={action}
 						controller={this.controller}
 					/>
 				);
+			} else if (action.actionType === ActionType.CUSTOM) {
+				const propertyId = this.controller.convertPropertyId(action.name);
```

Comment:
We usually use ControlUtil.getDataId for this.

Reply:
`getDataId()` expects a `propertyId`. So I didn't change line 52 and added `data-id={getDataId(propertyId)` on the div.